grpc/src/state_change_proxy.rs

```diff
@@ -14,69 +14,92 @@
 
 use crate::ankaios_streaming::GRPCStreaming;
 use crate::grpc_middleware_error::GrpcMiddlewareError;
-use api::proto;
-use api::proto::state_change_request::StateChangeRequestEnum;
-use api::proto::UpdateStateRequest;
+use api::proto::request::RequestContent;
+use api::proto::to_server::ToServerEnum;
+use api::proto::{self, Request};
+use api::proto::{RequestCompleteState, UpdateStateRequest};
 
 use common::request_id_prepending::prepend_request_id;
-use common::state_change_interface::{
-    StateChangeCommand, StateChangeInterface, StateChangeReceiver,
-};
+use common::state_change_interface::{StateChangeInterface, StateChangeReceiver, ToServer};
 
 use tokio::sync::mpsc::Sender;
 use tonic::Streaming;
 
 use async_trait::async_trait;
 
 pub struct GRPCStateChangeRequestStreaming {
-    inner: Streaming<proto::StateChangeRequest>,
+    inner: Streaming<proto::ToServer>,
 }
 
 impl GRPCStateChangeRequestStreaming {
-    pub fn new(inner: Streaming<proto::StateChangeRequest>) -> Self {
+    pub fn new(inner: Streaming<proto::ToServer>) -> Self {
         GRPCStateChangeRequestStreaming { inner }
     }
 }
 
 #[async_trait]
-impl GRPCStreaming<proto::StateChangeRequest> for GRPCStateChangeRequestStreaming {
-    async fn message(&mut self) -> Result<Option<proto::StateChangeRequest>, tonic::Status> {
+impl GRPCStreaming<proto::ToServer> for GRPCStateChangeRequestStreaming {
+    async fn message(&mut self) -> Result<Option<proto::ToServer>, tonic::Status> {
         self.inner.message().await
     }
 }
 
 // [impl->swdd~grpc-agent-connection-forwards-commands-to-server~1]
 pub async fn forward_from_proto_to_ankaios(
     agent_name: String,
-    grpc_streaming: &mut impl GRPCStreaming<proto::StateChangeRequest>,
-    sink: Sender<StateChangeCommand>,
+    grpc_streaming: &mut impl GRPCStreaming<proto::ToServer>,
+    sink: Sender<ToServer>,
 ) -> Result<(), GrpcMiddlewareError> {
     while let Some(message) = grpc_streaming.message().await? {
         log::trace!("REQUEST={:?}", message);
 
         match message
-            .state_change_request_enum
+            .to_server_enum
             .ok_or(GrpcMiddlewareError::ReceiveError(
                 "Missing state_change_request".to_string(),
             ))? {
-            StateChangeRequestEnum::UpdateState(UpdateStateRequest {
-                new_state,
-                update_mask,
+            ToServerEnum::Request(Request {
+                request_id,
+                request_content,
             }) => {
-                log::debug!("Received UpdateStateRequest from {}", agent_name);
-                match new_state.unwrap_or_default().try_into() {
-                    Ok(new_state) => {
-                        sink.update_state(new_state, update_mask).await?;
+                log::debug!("Received Request from {}", agent_name);
```

Comment:
The trace message should surround variables with simple quotes.
See below too.

Reply:
fixed